packages/react/src/Details/Details.stories.tsx

```diff
@@ -12,7 +12,7 @@ export const Default: StoryFn<typeof Details> = () => {
   const {getDetailsProps} = useDetails({closeOnOutsideClick: true})
   return (
     <Details {...getDetailsProps()}>
-      <Button as="summary">See Details</Button>
+      <Details.Summary as={Button}>See Details</Details.Summary>
```

Comment:
Couple noob questions here:

1. Assuming this renders a `<button>`, is that still accessible with `<details>`?
2. Just checking, does this also support ButtonProps like `size=small` and variant=primary`?

Reply:
1. Note sure what accessible with details means 😅, here's how/what it renders: 
<img width="1389" alt="image" src="https://github.com/user-attachments/assets/774e413c-ca7e-49cb-8c53-4a395dea4af1">
is there some specific behavior you want me to test for here?


2. Had to fiddle with the types a little but yeah those work: 
<img width="501" alt="image" src="https://github.com/user-attachments/assets/6f24245d-2511-468f-b21c-b92083891ac2">

<img width="1246" alt="image" src="https://github.com/user-attachments/assets/24468ff1-446a-4e90-ae1b-b6a513c068d6">